app/src/main/java/programatorus/client/screens/debugger/DebuggerFragment.kt

```diff
@@ -34,16 +51,41 @@ class DebuggerFragment : Fragment() {
             sendCommand()
         }
     }
+
+    private fun startLoop() {
+        future = CompletableFuture()
+        loop()
+    }
+
+    private fun loop() {
+        device.pollDebuggerLine().thenAccept { line ->
+            Log.d(TAG, "line polled: $line")
+            receive(line)
+            loop()
+        }
```

Comment:
future never completes

Reply:
Fixed in https://github.com/bgrzesik/programatorus-server/pull/18